packages/widget/src/pages/SwapExecutionPage/SwapExecutionButton.tsx

```diff
@@ -33,6 +35,36 @@ export const SwapExecutionButton: React.FC<SwapExecutionButtonProps> = ({
   connectRequiredChains,
   submitExecuteRouteMutation,
 }) => {
+  const [countdown, setCountdown] = useState<number | undefined>(
+    route?.estimatedRouteDurationSeconds,
+  );
+  const [timer, setTimer] = useState<ReturnType<typeof createCountdownTimer> | undefined>();
+
+  useEffect(() => {
+    const estimatedDurationSeconds = route?.estimatedRouteDurationSeconds;
+    if (
+      !timer &&
+      estimatedDurationSeconds &&
+      swapExecutionState &&
+      [SwapExecutionState.pending, SwapExecutionState.signaturesRemaining].includes(
+        swapExecutionState,
+      )
+    ) {
+      const countdownTimer = createCountdownTimer({
+        duration: estimatedDurationSeconds,
+        onUpdate: (remainingTime) => {
+          setCountdown(parseInt(remainingTime.toString()));
+        },
+      });
+      setTimer(countdownTimer);
+      countdownTimer.startCountdown();
+    }
+
+    return () => {
+      timer?.stopCountdown();
+    };
+  }, [route?.estimatedRouteDurationSeconds, swapExecutionState, timer]);
```

Comment:
can we extract this into a custom hook `useCountdown`
```
const {countdown} = useCountdown({
estimatedRouteDurationSeconds, 
swapExecutionState
})
```

Reply:
done